spinalcordtoolbox/image.py

```diff
@@ -914,9 +914,6 @@ def concat_data(im_in_list: Sequence[Image], dim, pixdim=None, squeeze_data=Fals
     if im_in_first.absolutepath is not None:
         im_out.absolutepath = add_suffix(im_in_first.absolutepath, '_concat')
 
-    # write file
-    im_out.data = data_concat
-
```

Comment:
NB: While making the above change, I noticed that this statement is redundant with the section immediately below:

https://github.com/neuropoly/spinalcordtoolbox/blob/d60106b8e01803cca081896304b5c6e88bb467e3/spinalcordtoolbox/image.py#L920-L925